mlflow/pyfunc/scoring_server/__init__.py

```diff
@@ -258,66 +213,49 @@ def transformation():  # pylint: disable=unused-variable
                 mimetype="text/plain",
             )
 
-        content_format = parameter_values.get("format")
-
+        unexpected_content_parameters = set(parameter_values.keys()).difference(set(["charset"]))
+        if unexpected_content_parameters:
+            return flask.Response(
+                response=(
+                    f"Unrecognized content type parameters: "
+                    f"{', '.join(unexpected_content_parameters)}."
+                ),
+                status=415,
+                mimetype="text/plain",
+            )
```

Comment:
Is the motivation for throwing on unrecognized type parameters to make it clear to users of 1.x workflows that the `format=pandas-records`, etc parameters are no longer supported?

Just making sure this is safe for existing use cases, i.e. is it possible that some users / orgs might want to pass custom content type parameters so that some upstream layer can do request logging / auditing or something?

Reply:
Yes that's exactly the intent - so we don't silently ignore stuff. Good question about the upstream - I would say we can start with being restrictive and loosen it in the future if we have to?